cli/jira.go

```diff
@@ -53,14 +54,27 @@ type JIRAClient interface {
 func NewJIRAClient(config *cfg.Config) (JIRAClient, error) {
 	log := config.GetLogger()
 
+	var oauth *http.Client
+	var err error
+	if config.GetConfigString("jira-user") == "" && config.GetConfigString("jira-pass") == "" {
+		oauth, err = getJIRAHTTPClient(*config)
+		if err != nil {
+			log.Errorf("Error getting OAuth config: %v", err)
+			return dryrunJIRAClient{}, err
+		}
+	}
+
 	var j JIRAClient
 
-	client, err := jira.NewClient(nil, config.GetConfigString("jira-uri"))
+	client, err := jira.NewClient(oauth, config.GetConfigString("jira-uri"))
 	if err != nil {
 		log.Errorf("Error initializing JIRA client; check your base URI. Error: %v", err)
 		return dryrunJIRAClient{}, err
 	}
-	client.Authentication.SetBasicAuth(config.GetConfigString("jira-user"), config.GetConfigString("jira-pass"))
+
+	if config.GetConfigString("jira-user") != "" || config.GetConfigString("jira-pass") != "" {
```

Comment:
This is a little funny because it's checking the opposite of line 59. Looks like this should really be an `else` clause. It will require a little reformatting and will require you to have two:
```go
client, err := jira.NewClient(...)
```
but I think it is a little easier to read. At the end of the day, both of the approaches will be equally performant, but I think this one will be more maintainable.